src/store/workflows.module.js

```diff
@@ -63,41 +49,519 @@ const state = {
 }
 
 const getters = {
-  currentWorkflow: state => {
-    if (state.workflowName === null) {
-      return null
+  /* Return matching nodes from the store.
+   *
+   * - Specify the type of node you want to find using nodeType.
+   * - Specify any IDs you want to filter for by specifying ids, or omit
+   *   this argument to return all.
+   *
+   * Note: This walks the tree from the root node down to the level specified
+   * by nodeType every time a node in the tree to this point is added or
+   * removed. E.G. if you request 'workflow' nodes, then this node list
+   * will be recomputed every time a workflow is added or removed (delta batching
+   * may reduce the actual call count).
+   */
+  getNodes: (state) => (nodeType, ids) => {
+    if (!NODE_TYPES.includes(nodeType)) {
+      throw new Error(`Invalid node type: ${nodeType}`)
+    }
+
+    // node types which are above "nodeType" in the tree
+    const parentNodeTypes = NODE_TYPES.slice(0, NODE_TYPES.indexOf(nodeType))
+
+    // walk the tree looking for nodeType nodes
+    let item
+    const ret = []
+    const stack = [...state.cylcTree.children]
+    while (stack.length) {
+      item = stack.pop()
+      if (parentNodeTypes.includes(item.type)) {
+        // this is above "nodeTyoe" in the tree, look through its child nodes
+        stack.push(...item.children)
+      } else if (
+        item.type === nodeType &&
+        (!ids || ids.includes(item.id))
+      ) {
+        ret.push(item)
+      }
     }
-    return Object.values(state.workflows)
-      .find(workflow => workflow.name === state.workflowName)
+    return ret
+  }
+}
+
+/* Initialise the data store. */
+function createTree (state) {
+  // console.log('@@ Create')
+  if (state.cylcTree) {
+    return
+  }
+  const tree = {
+    $index: {},
+    id: '$root'
+  }
+  Vue.set(tree, 'children', [])
```

Comment:
I think the reason we need to do this here is because Vue needs to hook into Array objects to make them reactive. So the object is fine, it's the thing we're adding to the object that we need to make reactive. I've got a feeling that this is different at Vue3.